src/components/uploader/layers/image.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import styles from '../uploader.module.css';
+import { UploaderEnum, useUploader } from '../../../reducers';
+
+export const UploaderLayerImage = () => {
+  const { uploader, dispatch } = useUploader();
+  const [configured, setConfigured] = useState(false);
+  const imageRef = useRef<HTMLImageElement>(null);
+  useEffect(() => {
+    if (!imageRef?.current || configured) {
+      return;
+    }
+    const { naturalWidth, naturalHeight, width, height } = imageRef.current;
+    dispatch({
+      type: UploaderEnum.SetOriginalImageWidth,
+      payload: naturalWidth,
+    });
+    dispatch({
+      type: UploaderEnum.SetOriginalImageHeight,
+      payload: naturalHeight,
+    });
+    dispatch({
+      type: UploaderEnum.SetStageWidth,
+      payload: width,
+    });
+    dispatch({
+      type: UploaderEnum.SetStageHeight,
+      payload: height,
+    });
+    dispatch({
+      type: UploaderEnum.SetRatio,
+      payload: width / naturalWidth,
+    });
+    console.log({ naturalWidth, naturalHeight, width, height });
+    setConfigured(!!naturalWidth);
+  });
```

Comment:
There is no list of dependencies. Is it intentional?

Reply:
Yes, for now, it's intentional. Because the image needs some time to be loaded and rendered. `naturalWidth` would take some time to be non-zero.
But I'll look into this and see if there's a way to refactor.